environment.py

```diff
@@ -49,42 +50,46 @@ def __init__(self, conf):
     
     def reset(self, options=None):
         ''' Choose initial state uniformly at random '''
-        self._state = np.zeros(self.conf.nb_state) 
+        state = np.zeros(self.conf.nb_state) 
 
-        rand_time = random.uniform(self.conf.x_init_min[-1], self.conf.x_init_max[-1])
+        time = random.uniform(self.conf.x_init_min[-1], self.conf.x_init_max[-1])
         for i in range(self.conf.nb_state-1): 
-            self._state[i] = random.uniform(self.conf.x_init_min[i], self.conf.x_init_max[i]) 
-        self._state[-1] = self.conf.dt*round(rand_time/self.conf.dt)
-        observation = self._state
+            state[i] = random.uniform(self.conf.x_init_min[i], self.conf.x_init_max[i]) 
+        state[-1] = self.conf.dt*round(time/self.conf.dt)
 
-        return rand_time, observation
+        return time, state
 
-    def step(self, rand_time, state, action):
+    def step(self, state, action):
         ''' Return next state and reward '''
         # compute next state
         state_next = self.simulate(state, action)
 
         # compute reward
-        reward = self.reward(rand_time, state_next, action)
+        reward = self.reward(state_next, action)
```

Comment:
How about here we switch to the standard TO formulation, using the current state to compute the reward? @gianluigigrandesso Currently I am not even sure that the two problems solved by TO and RL are exactly equivalent because of this difference.